From 45aec5a9d2d5c80a3f0dbb4da9ffba6891f72d6e Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:55 -0400
Subject: [infiniband] cxgb4: Update snd_seq when sending MPA messages

Message-id: <1410475447-94392-37-git-send-email-svemuri@redhat.com>
Patchwork-id: 93804
O-Subject: [RHEL7.1 PATCH BZ 1124947 036/108] RDMA/cxgb4: Update snd_seq when sending MPA messages
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 9c88aa003d26e9f1e9ea6e08511768c2ef666654)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index a9fb73a..76d4884 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -762,6 +762,7 @@ static void send_mpa_req(struct c4iw_ep *ep, struct sk_buff *skb,
  start_ep_timer(ep);
  state_set(&ep->com, MPA_REQ_SENT);
  ep->mpa_attr.initiator = 1;
+ ep->snd_seq += mpalen;
  return;
 }
 
@@ -841,6 +842,7 @@ static int send_mpa_reject(struct c4iw_ep *ep, const void *pdata, u8 plen)
  t4_set_arp_err_handler(skb, NULL, arp_failure_discard);
  BUG_ON(ep->mpa_skb);
  ep->mpa_skb = skb;
+ ep->snd_seq += mpalen;
  return c4iw_l2t_send(&ep->com.dev->rdev, skb, ep->l2t);
 }
 
@@ -925,6 +927,7 @@ static int send_mpa_reply(struct c4iw_ep *ep, const void *pdata, u8 plen)
  t4_set_arp_err_handler(skb, NULL, arp_failure_discard);
  ep->mpa_skb = skb;
  state_set(&ep->com, MPA_REP_SENT);
+ ep->snd_seq += mpalen;
  return c4iw_l2t_send(&ep->com.dev->rdev, skb, ep->l2t);
 }
 
-- 
1.7.1